src/SFA.DAS.EmployerIncentives.Domain/IncentiveApplications/IncentiveApplication.cs

```diff
@@ -49,11 +49,13 @@ public void Submit(DateTime submittedAt, string submittedByEmail, string submitt
 
             AddEvent(new Submitted
             {
-                 AccountId = AccountId,
-                 IncentiveApplicationId = Id,
-                 SubmittedAt = submittedAt,
-                 SubmittedBy = submittedByName,
-                 SubmittedByEmail = submittedByEmail
+                AccountId = AccountId,
+                IncentiveApplicationId = Id,
+                SubmittedAt = submittedAt,
+                SubmittedBy = submittedByName,
+                SubmittedByEmail = submittedByEmail,
+                AccountLegalEntityId = Model.AccountLegalEntityId,
+                Apprenticeships = Model.ApprenticeshipModels,
```

Comment:
This shouldn't be put on the domain event here.  The Submit method only knows about the submittedAt, submittedBy and submittedByEmail information.  You are coupling this domain event to another client.  That is what the event handlers are for.  AccountLegalEntityId  and Apprenticeships should't be in this domain event you have coupled a use case from the other domain to this aggregate.

Reply:
I don't think that's necessarily correct that it's coupling the event to the client.  It isn't unreasonable to expose what has been submitted on a submitted event.

Reply:
Why not just raise the event with the Model in that case? e.g.

 AddEvent(new Submitted   { IncentiveApplication= Model} ?

Then you get all the data in the event.

Reply:
sounds good to me :)